File: ftf.py
from random import *

listCity = [['Lome','Tsevie','Vogan','Aneho','Tabligbo'],
            ['Atakpame','Notse','Kpalime','Badou','Amlame'],
            ['Sokode','Tchamba','Sotouboua','Blitta','Djarkpanga'],
            ['Bafilo','Bassar','Kara', 'Kpagouda','Niamtougou','Defale'],
            ['Dapaong','Mango','Galangachi','Biankouri','Vavou']]

def choixVille(listCity):
  #for i in range (0,12):
    Region1=choice(listCity)
    Region2=choice(listCity)
    while Region1 ==Region2:
      Region1=choice(listCity)
      Region2=choice(listCity)

    while Region1==[]:
      listCity.remove(Region1)
      Region1=choice(listCity)

    while Region2==[]:
      listCity.remove(Region2)
      Region2=choice(listCity)

    else:
      playoffCity1 = choice(Region1)
      print ("Generating Playoff City 1:", playoffCity1)
      Region1.remove(playoffCity1)
      playoffCity2 = choice(Region2)
      print ("Generating Playoff City 2:", playoffCity2)
      Region2.remove(playoffCity2)
  # return  print("All cities have been selected")

def selection(Region):
  listCity1 = ['Lome','Tsevie','Vogan','Aneho','Tabligbo']
  listCity2 = ['Atakpame','Notse','Kpalime','Badou','Amlame']
  listCity3 = ['Sokode','Tchamba','Sotouboua','Blitta','Djarkpanga']
  listCity4 = ['Bafilo','Bassar','Kara', 'Kpagouda','Niamtougou','Defale']
  listCity5 = ['Dapaong','Mango','Galangachi','Biankouri','Vavou']

  if Region == 'Maritime':
    playoffCity = choice(listCity1)
    print (playoffCity)
  elif Region == 'Plateaux':
    playoffCity = choice(listCity2)
    print (playoffCity)
  elif Region == 'Centrale':
    playoffCity = choice(listCity3)
    print (playoffCity)
  elif Region == 'Kara':
    playoffCity = choice(listCity4)
    print (playoffCity)
  elif Region == 'Savanes':
    playoffCity = choice(listCity5)
    print (playoffCity)
  else :
    print ("Please enter a valid region") 

def semi_final(Region1,Region2):
  return (selection(Region1),selection(Region2))
  
def final(Region):
  return selection(Region)


menu=True
while menu:
    print ("""
        !!!WELCOME TO THE TOURNAMENT!!!
              1. Enter game type
              2. Exit
          """)
    menu=input ("WELCOME! Please choose: ")
    if menu== "1":
            gameType = input("Enter Game type:")
            while (gameType !="Kpessekou" and gameType !="Zobibi") :
              print ("Invalid input")
              gameType = input("Enter Game type:")
            if gameType=="Kpessekou":
                choixVille(listCity)
            elif gameType== "Zobibi":
                print ("""1.Semi-Final    2.Final""")
                menu2=input("Please choose: ")
                if menu2=="1":
                  Region1 = input('Enter Region 1:' )
                  Region2 = input('Enter Region 2:' )
                  while Region2 == Region1:
                     Region2 = input("Enter a Different Region: ")
                  if Region2 != Region1:
                    semi_final(Region1,Region2)
                if menu2=="2":
                  print ("Enter Region : ")
                  Region = input()
                  final(Region)

    elif menu=="2":
            print("Goodbye")
            break   

    else:
            print("Not Valid Choice Try again")

File: ftf.js
const prompt = require('prompt-sync')();
var country = {
    "Maritime": ['Lome', 'Tsevie', 'Vogan', 'Aneho', 'Tabligbo'],
    "Plateaux": ['Atakpame', 'Notse', 'Kpalime', 'Badou', 'Amlame'],
    "Centrale": ['Sokode', 'Tchamba', 'Sotouboua', 'Blitta', 'Djarkpanga'],
    "Kara": ['Bafilo', 'Bassar', 'Kara', 'Kpagouda', 'Niamtougou'],
    "Savanes": ['Dapaong', 'Mango', 'Galangachi', 'Biankouri', 'Vavou']
}
var region, region1, region2, city, city1, city2, gametype;

function kpessekou() {
    region1 = Object.keys(country)[Math.floor(Math.random() * Object.keys(country).length)];
    region2 = Object.keys(country)[Math.floor(Math.random() * Object.keys(country).length)]

    city1 = Object.values(country[region1])[Math.floor(Math.random() * Object.values(country[region1]).length)];
    city2 = Object.values(country[region2])[Math.floor(Math.random() * Object.values(country[region2]).length)];
    console.log("PlayoffCity1: " +city1);
    console.log("PlayoffCity2: " +city2);
}

function zobibi(region) {
    switch (region) {
        case 'Maritime':
            city = Object.values(country["Maritime"])[Math.floor(Math.random() * Object.values(country["Maritime"]).length)];
            console.log(city);
            break;
        case 'Plateaux':
            city = Object.values(country["Plateaux"])[Math.floor(Math.random() * Object.values(country["Plateaux"]).length)];
            console.log(city);
            break;
        case 'Centrale':
            city = Object.values(country["Centrale"])[Math.floor(Math.random() * Object.values(country["Centrale"]).length)];
            console.log(city);
            break;
        case 'Kara':
            city = Object.values(country["Kara"])[Math.floor(Math.random() * Object.values(country["Kara"]).length)];
            console.log(city);
            break;
        case 'Savanes':
            city = Object.values(country["Savanes"])[Math.floor(Math.random() * Object.values(country["Savanes"]).length)];
            console.log(city);
            break;

        default:
            do {
                region = prompt('Please enter Region : ');
            }
            while (!Object.keys(country).includes(region));
            break;
    }
}

function semiFinal(region1, region2) {
    return (zobibi(region1), zobibi(region2))
}

function final(region) {
    return (zobibi(region))
}

let menu = true;
while (menu) {
    console.log('\n !!!WELCOME TO THE TOURNAMENT!!! \n      1. Enter game type \n      2. Exit \n');
    menu = prompt("WELCOME! Please choose: ");
    if (menu === "1") {
        gametype = prompt("Enter game type: ");
        while (gametype !== "Kpessekou" && gametype !== "Zobibi") {
            console.log("Invalid input ");
            gametype = prompt("Enter game type: ");
        }
        if (gametype === "Kpessekou") {
            kpessekou()
        }
        else if (gametype === "Zobibi") {
            console.log("1. Semi-Final \n2. Final \n");
            menu = prompt("Please choose: ");
            if (menu === "1") {
                region1 = prompt("Enter Region 1:" );
                while(!Object.keys(country).includes(region1)){
                    region1 = prompt("Enter Correct Region 1:" );
                }
                region2 = prompt("Enter Region 2:" );
                while(!Object.keys(country).includes(region2)){
                    region2 = prompt("Enter Correct Region 2:" );
                }
                if (region2 === region1) {
                    region2 = prompt("Enter a Different Region: ")
                }
                else {
                    semiFinal(region1, region2);
                }
            }
            if (menu === "2") {
                region = prompt("Enter Final Region :");
                while(!Object.keys(country).includes(region)){
                    region = prompt("Enter Final Region :");
                }
                final(region)
            }
        }
    }
    else if (menu === "2") {
        console.log("Goodbye!!!");
        break;
    }
    else {
    }
}
